[core]/es_extended/shared/functions.lua

```diff
@@ -170,4 +170,43 @@ function ESX.IsFunctionReference(val)
     local typeVal = type(val)
 
     return typeVal == "function" or (typeVal == "table" and type(getmetatable(val)?.__call) == "function")
+end
+
+---@param conditionFunc function A function that returns a boolean indicating whether the condition is met.
+---@param timeout number The maximum time (in milliseconds) to wait for the condition to be met.
+---@param errorMessage? string The error message to print if the condition is not met within the timeout period.
+---@return boolean: Returns true if the condition is met within the timeout, otherwise returns `false`.
+---@return number: The time (in milliseconds) it took to meet the condition, or the timeout time if not met.
+ESX.Await = function(conditionFunc, timeout, errorMessage)
```

Comment:
Use `ESX.IsFunctionReference` to validate conditionFunc.

Personally, I'd also prefer to have `errorMessage` as the second param. More often than not, we will want to provide an error message without increasing the default timeout. Makes it a little easier to work with this function.

Reply:
Do you think the function should provide a default error message, or just not print anything if it doesnt hit the condition within the `timeout` param?

Reply:
I think we can leave that up to the caller. Surely there are usecases where we want to await for an action with the expectation that it may fail. I don't think we should throw an error in that case. However, if an errorMessage is provided, we should throw an actual error.